src/utils/io.js

```diff
@@ -6,13 +6,31 @@
 import axios from "axios";
 
 const saveBlob = async (data, name, type) => {
-  const link = document.createElement('a');
-  link.style.display = 'none';
-  document.body.appendChild(link);
   const blob = new Blob([data], { type });
-  link.href = URL.createObjectURL(blob);
-  link.download = name;
-  link.click();
+  // No foolproof way to detect node. So try loading what you require.
```

Comment:
In certain places we check `typeof window === undefined`.  Would that be `true` in electron?  As in, it has a global window variable but also has node?

Reply:
In electron `window` is defined in a renderer process (the browser tab). 
Also in any environment, `var window = "something"` will cause all kinds of issues. 

Detecting node is fraught with issues. Check this:
[https://stackoverflow.com/questions/34550890/how-to-detect-if-script-is-running-in-browser-or-in-node-js](https://stackoverflow.com/questions/34550890/how-to-detect-if-script-is-running-in-browser-or-in-node-js)
Specially the comments below the answers to see the edge cases where it won't work. 
So I've always chosen to detect the availability of the features that you want to use. And if those features are available, the code will succeed, otherwise fall back to default functionality. This is foolproof in this context. In a different context, a different detection appropriate to that context would work.